mlir/include/mlir/Dialect/LLVMIR/LLVMDialect.td

```diff
@@ -58,6 +58,10 @@ def LLVM_Dialect : Dialect {
     /// effect when lowering to the LLVMDialect.
     static StringRef getReadnoneAttrName() { return "llvm.readnone"; }
 
+    /// Name of the helper attribute to keep GPU workgroup attribution size
+    /// information when converting from GPU to LLVM.
+    static StringRef getWorkgroupAttribSizeAttrName() { return "llvm.mlir.workgroup_attrib_size"; }
```

Comment:
Can you use the:
```
      let discardableAttrs = (ins
         "mlir::IntegerAttr":$llvm.mlir.workgroup_attrib_size,
      );
```

Construction instead? This provides a type-safe helper to manipulate discardable attributes defined by dialects.

Reply:
This is to be used *only* with function argument attributes. As that won't help here, I'd stick with the simpler method. It's good to know that exists, tho, for future usages. Thanks!

Reply:
It helps in that it caches automatically a StringAttr on the Dialect instead. You can cache it manually if you prefer, but StringRef-based APIs are inefficient and we should just avoid these in general IMO.